src/app/history/recorder.js

```diff
@@ -30,14 +31,13 @@ export default class Recorder {
   }
 
   async save({ serviceId, documentType, content, fileExtension }) {
-    const directory = `${this.path}/${serviceId}`;
+    const filePath = this.getPathFor(serviceId, documentType, fileExtension);
```

Comment:
I went with tweaking a bit the `Recorder` to be able to reuse all of its logic even for storing files at the root of the versions repository (hence not belonging to a particular service provider).

As a result, I noticed a logical duplication in the code here, computing the path twice. I edited this part to actually first compute the full path and then ensuring required folder exists through `dirname`.